test/http.cc

```diff
@@ -12,9 +13,115 @@ using eventuals::Terminate;
 
 using eventuals::http::Get;
 using eventuals::http::Post;
-using eventuals::http::PostFields;
 
-class HTTPTest
+class HttpMockServer {
+ public:
+  HttpMockServer() {
+    PCHECK((server_ = socket(AF_INET, SOCK_STREAM, 0)) >= 0)
+        << "Failed to call 'socket()'";
+
+    address_.sin_family = AF_INET;
+    address_.sin_port = 0;
+    address_.sin_addr.s_addr = INADDR_ANY;
+
+    socklen_t length = sizeof(address_);
+
+    PCHECK(bind(server_, (sockaddr*) &address_, length) >= 0)
+        << "Failed to call 'bind()'";
+
+    PCHECK(listen(server_, 1) == 0)
+        << "Failed to call 'listen()'";
+
+    PCHECK(getsockname(server_, (sockaddr*) &address_, &length) == 0)
+        << "Failed to call 'getsockname()'";
+
+    ON_CALL(*this, Accepted)
+        .WillByDefault([this](int client) {
+          // Receive data up to end of headers.
+          std::string data;
+
+          do {
+            char buffer[1024];
+            size_t bytes = recv(client, buffer, sizeof(buffer), 0);
+            if (bytes <= 0) {
+              PLOG(ERROR) << "Failed to receive headers";
+              close(client);
```

Comment:
We check return values in each and every other socket calls, but we are not doing it for `close()`, why?